### PR TITLE
Node.js message routing update

### PR DESCRIPTION
**Pull request by carlosscastro**
- [x] Is numbered and named correctly and following the spec
--
- [ ] All files have been spell checked
- [ ] All style cop/ linting issues have been addressed
- [x] Include all DCR changes
- [x] Explicit state save
- [x] Method name changes begin, end etc
- [x] Does not use mainDialog unless that file actually extends Dialog/ ComponentDialog
- [x] Compiles with latest packages
- [x] Runs per instructions in README
- [x] Add commented out code to use blob storage
- [x] Update to look for production/ local (bot configuration)
- [ ] onTurnError at adapter
- [x] Include .env or appsettings.json or web.config with botFilePath and botFileSecret
- [x] Deployment scripts folder has been cleaned up and only includes msbot export files
- [ ] README to include
 
- [x] msbot clone steps
- [ ] Tools install/ uninstall
- [x] Concise set of steps to run the sample
- [ ] C# sample -- Include .bot file in the build output
- [ ] All samples include the latest botfile (no "secretKey" field; replaced with "padLock"). Use latest MSBOT off the tools MyGet feed to generate the bot file
- [ ] remove and sort usings
- [x] remove azuredeploy.* under deploymentscripts folder (if any)
- [x] Welcoming users is consistent & correct. See 13.basic bot for nodejs & 3. welcome user for C#

### CODE REVIEW

**Comment by carlosscastro** [comment]
@vishwacsena  I was planning on putting moving it in another PR, not sure if I should move to samples in this pr.

File: javascript_nodejs/09.message-routing/index.js [hunk 1]
```diff
@@ -79,10 +79,6 @@ const memoryStorage = new MemoryStorage();
 const conversationState = new ConversationState(memoryStorage);
 const userState = new UserState(memoryStorage);
 
-// Use the BotStateSet middleware to automatically read and write conversation and user state.
```
Comment (vishwacsena) [inline javascript_nodejs/09.message-routing/index.js]: You might want to incldue onTurnError handler. See the nlp-with-dispatch bot for the correct code.

Comment (sgellock) [reply javascript_nodejs/09.message-routing/index.js]: I'd also move it into the samples folder.  I'll merge it once those two things are addressed

Comment (carlosscastro) [reply javascript_nodejs/09.message-routing/index.js]: Thanks for reviewing! Updating the PR in a few minutes :)

**Pull request merged**